apps/block_scout_web/assets/js/lib/analytics.js

```diff
@@ -0,0 +1,257 @@
+
+import $ from 'jquery'
+import Analytics from 'analytics'
+import segmentPlugin from '@analytics/segment'
+import omit from 'lodash/omit'
+import uniqid from 'uniqid'
+
+let analytics
+let store
+
+const initialState = {
+  userID: localStorage.getItem('userID')
+}
+
+function reducer (state = initialState, action) {
+  switch (action.type) {
+    case 'PAGE_LOAD':
+    case 'ELEMENTS_LOAD': {
+      return Object.assign({}, state, omit(action, 'type'))
+    }
+    case 'SET_USER_ID': {
+      const id = uniqid()
+      localStorage.setItem('userID', id)
+      return Object.assign({}, state, { userID: id })
+    }
+    default:
+      return state
+  }
+}
+
+// return string representation of web page path
+function getPageName (path) {
+  switch (true) {
+    case path.includes('/search'):
+      return '404SearchResult'
+    case path === '/':
+      return 'home'
+    case path === '/txs':
+      return 'validatedTransactions'
+    case path === '/pending_transactions':
+      return 'pendingTransactions'
+    case path === '/blocks':
+      return 'blockHistory'
+    case path === '/accounts':
+      return 'allAccounts'
+    case path.includes('/blocks') && path.includes('/transactions'):
+      return 'blockTransactions'
+    case path.includes('/blocks') && path.includes('/signers'):
+      return 'blockSigners'
+    case path.includes('/address') && path.includes('/transactions'):
+      return 'addressDetails'
+    case path.includes('/address') && path.includes('/token_transfers'):
+      return 'addressTokenTransfers'
+    case path.includes('/address') && path.includes('/tokens'):
+      return 'addressTokens'
+    case path.includes('/address') && path.includes('/internal_transactions'):
+      return 'addressInternalTransactions'
+    case path.includes('/address') && path.includes('/coin_balances'):
+      return 'addressCoinHistory'
+    case path.includes('/address') && path.includes('/logs'):
+      return 'addressLogs'
+    case path.includes('/address') && path.includes('/signed'):
+      return 'addressDowntime'
+    case path.includes('/address') && path.includes('/validations'):
+      return 'addressBlocksValidated'
+    case path.includes('/address') && path.includes('/celo'):
+      return 'addressCeloInfo'
+    // TODO: Add rest of contract pages
+    case path.includes('/address') && path.includes('/contracts'):
+      return 'contractAddressCode'
+    case path.includes('/address') && path.includes('/read_contract'):
+      return 'readContract'
+    case path.includes('/tx') && path.includes('/internal_transactions'):
+      return 'transactionInternalTransaction'
+    case path.includes('/tx') && path.includes('/logs'):
+      return 'transactionLogs'
+    case path.includes('/tx') && path.includes('/raw_trace'):
+      return 'transactionRawTrace'
+    case path.includes('/tx') && path.includes('/token_transfers'):
+      return 'transactionTokenTransfers'
+    default:
+      return 'unknown'
```

Comment:
why not have default be the original path? that way we can see what we missed

Reply:
Ooh, good idea